lambdas/cmr-query/handler.py

```diff
@@ -1,11 +1,102 @@
+                {
+                    'collection': 'AfriSAR_UAVSAR_Ungeocoded_Covariance',
+                    'remote_fileurl': 's3://nasa-maap-data-store/file-staging/nasa-map/AfriSAR_UAVSAR_Ungeocoded_Covariance___1/uavsar_AfriSAR_v1-cov_topo_a41_r9_localPsi.rdr',
+                    'granule_id': 'G1200110696-NASA_MAAP',
+                    'id': 'G1200110696-NASA_MAAP',
+                    'mode': 'cmr',
+                    'test_links': None,
+                    'reverse_coords': None,
+                    'asset_name': 'data',
+                    'asset_roles': ['data'],
+                    'asset_media_type': 'application/x-hdr'
+                }
+            ]
+        )
+        Returns:
+        [
+            {
+                'collection': 'AfriSAR_UAVSAR_Ungeocoded_Covariance',
+                'remote_fileurl': 's3://nasa-maap-data-store/file-staging/nasa-map/AfriSAR_UAVSAR_Ungeocoded_Covariance___1/uavsar_AfriSAR_v1-cov_coreg_fine_hsixty_14050_16015_140_009_160308_cov_1-1.hdr',
+                'granule_id': 'G1200109928-NASA_MAAP',
+                'id': 'G1200109928-NASA_MAAP',
+                'mode': 'cmr',
+                'test_links': None,
+                'reverse_coords': None,
+                'asset_name': 'data',
+                'asset_roles': ['data'],
+                'asset_media_type': 'application/x-hdr',
+                'assets': {
+                    'cov_1-1.bin': 's3://nasa-maap-data-store/file-staging/nasa-map/AfriSAR_UAVSAR_Ungeocoded_Covariance___1/uavsar_AfriSAR_v1-cov_coreg_fine_hsixty_14050_16015_140_009_160308_cov_1-1.bin',
+                    'cov_1-1.hdr': 's3://nasa-maap-data-store/file-staging/nasa-map/AfriSAR_UAVSAR_Ungeocoded_Covariance___1/uavsar_AfriSAR_v1-cov_coreg_fine_hsixty_14050_16015_140_009_160308_cov_1-1.hdr',
+                    'cov_1-2.bin': 's3://nasa-maap-data-store/file-staging/nasa-map/AfriSAR_UAVSAR_Ungeocoded_Covariance___1/uavsar_AfriSAR_v1-cov_coreg_fine_hsixty_14050_16015_140_009_160308_cov_1-2.bin',
+                    ...,
+                }
+                'product_id': 'uavsar_AfriSAR_v1-cov_coreg_fine_hsixty_14050_16015_140_009_160308'
+            }
+        ]
+    """
+    fileurls_pattern = re.compile(data_file_regex)
+    objects = []
+    product_ids = {}
+
+    def _get_asset_name(remote_fileurl: str, product_id: str) -> str:
+        return re.sub(f".*{product_id}[-_.]?", "", remote_fileurl)
+
+    # Creates a Dict[product_id, Dict[file_name, List[str]]]
+    for item in data:
+        match = re.search(fileurls_pattern, item["remote_fileurl"])
+        if match:
+            product_id = match.group()
+            product_ids[product_id] = product_ids.get(product_id, {})
+
+            product_ids[product_id][
+                _get_asset_name(item["remote_fileurl"], product_id)
+            ] = item["remote_fileurl"]
+
+    # Creates an objects Dict of modified file_obj's, adding file_obj["assets"]
+    for product_id in product_ids.keys():
```

Comment:
Can you explain a bit more about what this code is doing and why it's needed? This should be a non-blocker, just asking for my own understanding.

Reply:
Yeah! Sorry if the comments weren't clear enough, but essentially we're grouping all the files with the same product_id under a `Dict[product_id, Dict[file_name, List[str]]]` and then adding those grouped files `List[str]` to the granule that we want to build the STAC record around (`data_file`) and adding it to a different `List[Dict]` for the `return_obj`